core/cafe_service.go

```diff
@@ -615,6 +618,28 @@ func (h *CafeService) handleRegistration(pid peer.ID, env *pb.Envelope) (*pb.Env
 		return h.service.NewError(403, errForbidden, env.Message.RequestId)
 	}
 
+	// does the provided token match?
+	s := strings.Split(reg.Token, "+")
+	if len(s) != 2 {
+		h.service.NewError(403, errForbidden, env.Message.RequestId)
```

Comment:
Could probably create a more nuanced error msg here? But this seemed sufficient?